nrf52-hal-common/src/timer.rs

```diff
@@ -146,4 +146,9 @@ impl<T> Timer<T> where T: TimerExt {
 
         Ok(())
     }
+
+    pub fn delay(&mut self, cycles: u32) {
+        self.start(cycles);
+        block!(self.wait()).unwrap();
```

Comment:
TODO: Should this result be returned?

Reply:
I don't think there is any use to this, no?

Reply:
I think the unwrap will only occur if something other than Error::WouldBlock is returned. That isn't possible today, but might be later? I'm not too familiar with the `nb` crate.

Reply:
Well I guess that would need be adressed when there changes anything about this. It makes no sense to have the user always write

`let _  = delay(42);`

imo.